packages/suite/src/components/firmware/CheckSeed.tsx

```diff
@@ -0,0 +1,99 @@
+import React from 'react';
+import styled from 'styled-components';
+
+import { Button, Checkbox, colors } from '@trezor/components';
+import { useDevice, useFirmware } from '@suite-hooks';
+import { Translation } from '@suite-components';
+import { P, H2, WarningImg, SeedImg, ReconnectInNormalStep } from '@firmware-components';
+
+const CheckboxRow = styled.div`
+    display: flex;
+    flex-direction: row;
+    border-top: 1px solid ${colors.NEUE_STROKE_GREY};
+    width: 380px;
+    padding: 20px 0;
+    justify-content: center;
+    margin: 28px auto 0 auto;
+`;
+
+const Body = () => {
+    const { device } = useDevice();
+    const { toggleHasSeed, hasSeed } = useFirmware();
+
+    // ensure that device is connected in requested mode
+    if (device?.mode !== 'normal') return <ReconnectInNormalStep.Body />;
+
+    if (!device?.features?.needs_backup) {
+        return (
+            <>
+                <SeedImg />
+                <H2>
+                    <Translation id="TR_SECURITY_CHECKPOINT_GOT_SEED" />
+                </H2>
+                <P>
+                    <Translation id="TR_BEFORE_ANY_FURTHER_ACTIONS" />
+                </P>
+                <CheckboxRow>
+                    <Checkbox
+                        isChecked={hasSeed}
+                        onClick={toggleHasSeed}
+                        data-test="@firmware/confirm-seed-checkbox"
+                    />
+                    <P>
+                        {' '}
+                        <Translation id="FIRMWARE_USER_HAS_SEED_CHECKBOX_DESC" />{' '}
+                    </P>
+                </CheckboxRow>
+            </>
+        );
+    }
+
+    if (device?.features?.needs_backup) {
```

Comment:
This condition is not needed as the opposite condition is checked above. I would even suggest keeping the true condition, remove the false condition and move the return below.

Reply:
With trezor.features I got used to write it as explicit possible. And even that did not save me from leaving a "bug" here! There is another field `unfinished_backup` that is set to `true` if you failed to create your backu (device disconnected during action). I ll rework it a bit here.